pkg/serviceaccount/claims.go

```diff
@@ -77,7 +84,16 @@ func Claims(sa core.ServiceAccount, pod *core.Pod, secret *core.Secret, expirati
 			UID:  string(secret.UID),
 		}
 	}
-	return sc, pc
+
+	// Tweak expiry fields for safe transition of projected service account token.
+	// Warn (instead of fail) after requested expiration time.
+	// Fail after hard-coded extended expiration time.
+	if extendExpiry && expirationSeconds == expirationSecondsSpecial {
```

Comment:
only do this for pod-bound tokens (if pod is non-nil), and if the pod has a projected token volume with the special expiry?

Reply:
I'm OK with checking whether it is pod-bound token. 
I thought our new projected tokens are time-bound and pod-bound at the same time, so if expirationSeconds are requested it's enough to say it's a new projected token? Am I misunderstanding something?